### PR TITLE
docs: add transport layer docs

### LINKED ISSUE #38: Architecture (post proposal): Transport layer
- The UI uses GraphQL - why, and what libraries are used to support this?
- Define/document the transport (WS, HTTP etc) used by the UI, and what they are used for

### PR DESCRIPTION
**Pull request by JQrdan**
Closes: #38

Signed-off-by: Jordan <jordan.tucker1@ibm.com>

### CODE REVIEW

File: docs/Architecture.md [hunk 1]
```diff
@@ -93,7 +93,21 @@ Further details will be added to this section once https://github.com/strimzi/pr
...
+
+#### HTTP
+
+Currently, HTTP is only used to retrieve static assets from the UI server.
+
+#### WebSockets
+
+[GraphQL](https://graphql.org/)
+
+All messages with GraphQL are done over a WebSocket, even standard queries. WebSockets are required by GraphQL to do subscriptions. Subscriptions allow the client to listen for data changes from the server so that it can stay updated in real-time. All other quieries are bundled into the same websocket so a clean network traffic log can be kept. The WebSocket will be held with the UI server, which will then proxy any websocket messages to the GraphQL admin server. This proxy allows for custom Apollo errors to be injected into the websocket so that the client can handle additional errors such as session expiration. The proxy is also responsible for injecting an authorization header for the client so that a connection can be etablished to the admin server. For full details of the WebSocket flow, [view our proposal](https://github.com/strimzi/proposals/blob/master/011-strimzi-ui.md#session-management-and-authentication).
```
Comment (ajborley) [inline docs/Architecture.md]: Typo: quieries -> queries

File: docs/Architecture.md [hunk 2]
```diff
@@ -93,7 +93,21 @@ Further details will be added to this section once https://github.com/strimzi/pr
...
+
+[Logging](#client-side-logging)
+
+Client logs are sent to the UI server over a persistant websocket so that the logs of both components can co-exist.
```
Comment (ajborley) [inline docs/Architecture.md]: Typo: persistant -> persistent

**Review by ajborley — APPROVED** [review]
LGTM - just a couple of typos I noticed.

File: docs/Architecture.md (hunk 1 above)
Comment (matthew-chirgwin) [inline docs/Architecture.md]: No objection to using websockets - will need to confirm if the server needs any change in the api module to proxy WS traffic

**Review by matthew-chirgwin — CHANGES_REQUESTED** [review]
Pending Andy's comments LGTM - re the use of websockets, is that something that can be checked under #55 ?

File: docs/Architecture.md (hunk 1 above)
Comment (JQrdan) [reply docs/Architecture.md]: Ok cool, yeah I'll use #55 to verify what needs to happen to the server to make this work

Comment (ajborley) [reply docs/Architecture.md]: My branch for #42 (https://github.com/ajborley/strimzi-ui/tree/42-client-side-logging) builds on Matt's experimental support for websocket handling:
https://github.com/ajborley/strimzi-ui/blob/42-client-side-logging/server/main.ts#L69-L79
https://github.com/ajborley/strimzi-ui/blob/42-client-side-logging/server/log/router.ts#L25-L34

**Review by matthew-chirgwin — APPROVED** [review]
LGTM

**Comment by matthew-chirgwin** [comment]
Any thoughts @dlabaj / @pmuir on the approach descirbed here?

**Review by nictownsend — APPROVED** [review]

File: docs/Architecture.md [hunk 3]
```diff
@@ -93,7 +93,21 @@ Further details will be added to this section once https://github.com/strimzi/pr
...
+
+#### HTTP
+
+Currently, HTTP is only used to retrieve static assets from the UI server.
+
+#### WebSockets
+
+[GraphQL](https://graphql.org/)
+
+All messages with GraphQL are done over a WebSocket, even standard queries. WebSockets are required by GraphQL to do subscriptions. Subscriptions allow the client to listen for data changes from the server so that it can stay updated in real-time. All other queries are bundled into the same websocket so a clean network traffic log can be kept. The WebSocket will be held with the UI server, which will then proxy any websocket messages to the GraphQL admin server. This proxy allows for custom Apollo errors to be injected into the websocket so that the client can handle additional errors such as session expiration. The proxy is also responsible for injecting an authorization header for the client so that a connection can be established to the admin server. For full details of the WebSocket flow, [view our proposal](https://github.com/strimzi/proposals/blob/master/011-strimzi-ui.md#session-management-and-authentication).
```
Comment (pmuir) [inline docs/Architecture.md:106]: It's definitely good to not mix http and ws.

**Review by pmuir — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Pull request merged**